assets/javascripts/discourse/components/chat-message-move-to-channel-modal-inner.js

```diff
@@ -0,0 +1,55 @@
+import Component from "@ember/component";
+import discourseComputed from "discourse-common/utils/decorators";
+import { isBlank } from "@ember/utils";
+import { action, computed } from "@ember/object";
+import { ajax } from "discourse/lib/ajax";
+import { inject as service } from "@ember/service";
+import { popupAjaxError } from "discourse/lib/ajax-error";
+
+export default class MoveToChannelModalInner extends Component {
+  @service chat;
+  @service router;
+  tagName = "";
+  sourceChannel = null;
+  destinationChannelId = null;
+  selectedMessageIds = null;
+
+  @computed("destinationChannelId")
+  get disableMoveButton() {
+    return isBlank(this.destinationChannelId);
+  }
+
+  @discourseComputed("chat.publicChannels.[]")
+  availableChannels(publicChannels) {
```

Comment:
Just curious, why do we define getters for the other computed properties but not this?

Reply:
this might also cause issues on legacy: discourseComputed in native classes is not working well, use computed please